common/configurable_model.py

```diff
@@ -0,0 +1,38 @@
+from bit_vector import BitVector
+from common.model import Model
+
+
+def ConfigurableModel(config_data_width, config_addr_width):
+    """
+    This function returns a class (parameterized by @config_data_width and
+    @config_addr_width) which can be used as the base class for any functional
+    model implementing a configurable module. For example if we have
+    configurable module Foo:
+
+        class Foo(ConfigurableModel(32, 32)):
+            ...
+
+    In our test code we can then do:
+
+        foo = Foo(...)
+        foo.configure(addr, data)
+        assert foo.read_config(addr) == data
+
+    Or internally in Foo, do:
+
+        conf = self._config[addr]
+
+    """
+    class _ConfigurableModel(Model):
+        def __init__(self):
+            self._config = {}
+
+        def configure(self, addr: BitVector, data: BitVector):
+            assert data.num_bits == config_data_width
+            assert addr.num_bits == config_addr_width
+            self._config[addr.unsigned_value] = data
+
+        def read_config(self, addr):
+            return self._config[addr.unsigned_value]
```

Comment:
What about using a `@property` to implement a getter/setter interface like
```
foo.config[addr] = data
```
and
```
conf = foo.config[addr]
```

Here's some docs on property: https://docs.python.org/3.6/library/functions.html#property